src/js/Controllers/UIController.js

```diff
@@ -490,60 +490,65 @@ class UIController {
-                      delete this.timers[app.scrollableMessage.msgID];
-                      this.listener.send(RpcFactory.ScrollableMessageAbortedResponse(app.scrollableMessage.msgID));
-                      store.dispatch(closeScrollableMessage(app.alert.msgID, rpc.params.appID));
-                      const context = getNextSystemContext();
-                      this.onSystemContext(context, state2.activeApp);
-                      return true;
                     }
+                    this.onSystemContext(context, state.activeApp)
+                    return true
+                } else if (rpc.params.functionID === 64 && app.alert.showAlert && app.alert.isSubtle
+                    && (rpc.params.cancelID === undefined || rpc.params.cancelID === app.alert.cancelID)) {
+                    clearTimeout(this.timers[app.alert.msgID])
+                    delete this.timers[app.alert.msgID]
+                    this.listener.send(RpcFactory.SubtleAlertErrorResponse(app.alert.msgID, 5, 'subtle alert was cancelled'))
+                    store.dispatch(closeAlert(app.alert.msgID, rpc.params.appID))
+                    const context = getNextSystemContext();
+                    if (rpc.params.appID !== state.activeApp) {
+                        this.onSystemContext("MAIN", rpc.params.appID)
+                    }
+                    this.onSystemContext(context, state.activeApp)
+                    return true
+                } else if (rpc.params.functionID === 26 && app.slider.showSlider
+                    && (rpc.params.cancelID === undefined || rpc.params.cancelID === app.slider.cancelID)) {
+                    clearTimeout(this.timers[app.slider.msgID])
+                    delete this.timers[app.slider.msgID]
+                    this.listener.send(RpcFactory.SliderAbortedResponse(app.slider.msgID))
+                    store.dispatch(closeSlider(app.alert.msgID, rpc.params.appID))
+                    const context = getNextSystemContext();
+                    this.onSystemContext(context, state.activeApp)
+                    return true
+                } else if (rpc.params.functionID === 25 && app.scrollableMessage.active
+                    && (rpc.params.cancelID === undefined || rpc.params.cancelID === app.scrollableMessage.cancelID)) {
+                    clearTimeout(this.timers[app.scrollableMessage.msgID]);
+                    delete this.timers[app.scrollableMessage.msgID];
+                    this.listener.send(RpcFactory.ScrollableMessageAbortedResponse(app.scrollableMessage.msgID));
+                    store.dispatch(closeScrollableMessage(app.alert.msgID, rpc.params.appID));
+                    const context = getNextSystemContext();
+                    this.onSystemContext(context, state.activeApp);
+                    return true;
                 }
                 
                 return { rpc: RpcFactory.UICancelInteractionIgnoredResponse(rpc) }
+            }
             case "ClosePopUp": {
```

Comment:
We'll probably need to edit the `OnSystemContext` usages here too

Reply:
Updated the calls from ClosePopUp in https://github.com/smartdevicelink/generic_hmi/pull/400/commits/d5533e893f784e9e88530620ac3df2c7ba27f089